libioc/Jail.py

```diff
@@ -563,41 +537,95 @@ def _stop_failed_jail(
                 )
             self._save_autoconfig()
 
-        try:
-            if single_command is None:
-                stdout, stderr, returncode = self._launch_persistent_jail(
-                    passthru=passthru
-                )
-            else:
-                stdout, stderr, returncode = self._launch_single_command_jail(
-                    single_command,
-                    passthru=passthru
+        if single_command is None:
+            # Start and Poststart Hooks
+            yield from self.__run_hook("start", False, jailStartEvent.scope)
+            yield from self.__run_hook(
+                "poststart",
+                False,
+                jailStartEvent.scope
+            )
+        else:
+            yield from self.__run_hook("command", False, jailStartEvent.scope)
+            yield from self.__destroy_jail(scope)
+            if jid is not None:
+                yield from self.__stop_network(
+                    jid,
+                    False,
+                    jailStartEvent.scope
                 )
-                scope = jailLaunchEvent.scope
-                jid = self.jid
-                yield from self.__destroy_jail(scope)
-                yield from self.__run_hook("poststop", False, scope)
-                if jid is not None:
-                    yield from self.__stop_network(jid, False, scope)
-                yield from self.__teardown_mounts(False, scope)
-                yield from self.__clear_resource_limits(False, scope)
-
-            if returncode != 0:
-                raise libioc.errors.JailLaunchFailed(
-                    jail=self,
+            yield from self.__teardown_mounts(False, jailStartEvent.scope)
+            yield from self.__clear_resource_limits(False, scope)
+            yield from self.__unmount_devfs(jailStartEvent.scope)
```

Comment:
Partially yes, but single-command execution and persistent jails require different hooks to be triggered. It is more error prone to repeat the other lines (because it can be easily forgotten on changes), but on the other hand reduces call stack complexity, which I find a good trade in this case.